templates/cws/cws_deployment.tmpl

```diff
@@ -0,0 +1,207 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app.kubernetes.io/component: portal-x
+    app.kubernetes.io/name: cws-test
+  name: cws-test
+  namespace: {{ .Namespace }}
+spec:
+  progressDeadlineSeconds: 600
+  replicas: 1
+  revisionHistoryLimit: 2
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: portal-x
+      app.kubernetes.io/name: cws-test
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/component: portal-x
+        app.kubernetes.io/name: cws-test
+        app: cws-test
+    spec:
+      containers:
+      - args:
+        - server
+        - --debug
+        - --dev
+        - --database
+        - $(DATABASE)
+        envFrom:
+        - secretRef:
+            name: customer-web-server-secret
+        image: mattermost/cws-test:{{ .ImageTag }}
+        imagePullPolicy: Always
+        name: cws-main
+        ports:
+        - containerPort: 8076
+          name: api
+          protocol: TCP
+        - containerPort: 8077
+          name: internal
+          protocol: TCP
+        resources: {}
+      imagePullSecrets:
+      - name: customer-web-server-gitlab-image
+      initContainers:
+      - args:
+        - schema
+        - migrate
+        - --database
+        - $(DATABASE)
+        envFrom:
+        - secretRef:
+            name: customer-web-server-secret
+        image: mattermost/cws-test:{{ .ImageTag }}
+        imagePullPolicy: Always
+        name: customer-web-server-init-database
+      restartPolicy: Always
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: customer-web-server-secret
+  namespace: {{ .Namespace }}
+type: Opaque
+stringData:
+  DATABASE: {{ .Environment.Database }}
+  CWS_PAYMENT_URL: {{ .Environment.CWSPaymentURL }}
+  CWS_PAYMENT_TOKEN: {{ .Environment.CWSPaymentToken }}
+  CWS_SITEURL: {{ .Environment.CWSSiteURL }}
+  CWS_SMTP_USERNAME: {{ .Environment.CWSSMTPUsername }}
+  CWS_SMTP_PASSWORD: {{ .Environment.CWSSMTPPassword }}
+  CWS_SMTP_SERVER: {{ .Environment.CWSSMTPServer }}
+  CWS_SMTP_PORT: "{{ .Environment.CWSSMTPPort }}"
+  CWS_SMTP_SERVERTIMEOUT: "{{ .Environment.CWSSMTPServerTimeout }}"
+  CWS_SMTP_CONNECTIONSECURITY: {{ .Environment.CWSSMTPConnectionSecurity }}
+  CWS_EMAIL_REPLYTONAME: {{ .Environment.CWSEmailReplyToName }}
+  CWS_EMAIL_REPLYTOADDRESS: {{ .Environment.CWSEmailReplyToAddress }}
+  CWS_EMAIL_BCCADDRESSES: {{ .Environment.CWSEmailBCCAddress }}
+  CWS_CLOUD_URL: {{ .Environment.CWSCloudURL }}
+---
+apiVersion: v1
+data:
+  .dockerconfigjson: {{ .Environment.DockerHubCredentials }}
```

Comment:
remember when you build this string it should be in base64 encoded

Reply:
Resolving this due to our discussion with DOCKER_HUB_CREDENTIALS variable - this will already be base64 encoded.